### ISSUE TITLE
Anisotropic interpolation (i.e. different specs in different dimensions)

### ISSUE THREAD

**Issue by timholy** [tlycken/Interpolations.jl#41]
I've encountered a situation where I want to interpolate along some dimensions but not others. I now wonder whether we should make the `IT` parameter of `BSplineInterpolation` a tuple, one per dimension of the array. One could use, for example, quadratic interpolation for dimensions 1-2 and linear interpolation along dimension 3, or skip interpolation along some dimensions. If this parameter is a tuple-parameter, then the `@generated` functions can generate the appropriate `getindex` code for the type.

Thoughts? We'd of course want there to be an "easy" interface that replicates current behavior, as well as one allowing more control.

**Comment by tlycken** [comment]
I've thought about this too - although I mostly thought about it in the context of boundary conditions and extrapolation. But yes, I think it would be a really nice and powerful feature to be able to specify different behavior in different dimensions.

I imagine a good solution would be to allow *either* arguments as we state them now, *or* tuples of them, and expand the single-argument calls to tuple calls before doing any actual handling of stuff. In other words,

```julia
A = rand(100, 100)
interpolate(A, Quadratic(Flat), Constant)
```

would be equivalent to

```julia
A = rand(100, 00)
interpolate(A, (Quadratic(Flat), Quadratic(Flat)), (Constant, Constant))
```

allowing for variations like

```
A = rand(100, 100)
interpolate(A; (Quadratic(Flat), None), (Constant, ErrorExtrap)```
interpolate(A, Quadratic((Flat,Line)), (Constant, Linear))
# etc...
```

Unfortunately, I won't have time to work on this in the foreseeable future, but I definitely like the idea so if you have a need for it and want to take a stab I'll gladly help out when I do find some time :)

**Comment by timholy** [comment]
Closed by #43
